src/indexers/PeriodicBackgroundIndexer.ts

```
@@ -0,0 +1,183 @@
+                });
+
+                await this.algoliaStore.periodicDataIndex.deleteObject(
+                  pkg.name
+                );
+
+                await this.mainIndex.deleteObject(pkg.name).wait();
+                return;
+              }
+            }
+          }
+
+          const npmDownloads = computeDownload(
+            pkg,
+            data.packageNpmDownloads,
+            data.totalNpmDownloads
+          );
+
+          const jsDelivrHits = jsDelivr.getHit(pkg);
+          const pkgPatch = {
+            objectID: pkg.objectID,
+            ...(npmDownloads || {}),
+            ...jsDelivrHits,
+            popular: npmDownloads?.popular || jsDelivrHits.popular,
+          };
+
+          patches.push({
+            ...pkgPatch,
+            _searchInternal: {
+              ...pkg._searchInternal,
+              popularAlternativeNames: getPopularAlternativeNames({
+                ...pkg,
+                ...pkgPatch,
+              }),
+            },
+            [this.facetField]: round(new Date(data.updatedAt)).valueOf(),
+          });
+        },
+        { concurrency: 20 }
+      );
+
+      await Promise.all([
+        this.algoliaStore.periodicDataIndex.saveObjects(dataIndexObjects),
+        this.mainIndex.partialUpdateObjects(patches).wait(),
+      ]);
+
+      datadog.increment('periodicDataIndex.success', task.pkg.length);
+    } catch (err) {
+      datadog.increment('periodicDataIndex.failure', task.pkg.length);
+      sentry.report(new Error(`Error in ${this.constructor.name}`), { err });
+
+      await this.mainIndex
+        .partialUpdateObjects(
+          task.pkg.map((pkg) => {
+            return {
+              objectID: pkg.objectID,
+              [this.facetField]: offsetToTimestamp(
+                ms('1 day'),
+                new Date(pkg[this.facetField])
+              ),
+            };
+          })
+        )
+        .wait()
+        .catch(() => {});
```

There is a lot of empty catches like this one, be careful not to silent all Algolia errors, they can happen more often than we think; 
SLOs are on search not indexing

---

Yes, I ignore the errors that don't impact the overall process and are generally non-actionable. E.g. here, it's a case when there's first an error in the indexing, then another error in the `catch` handler, and we failed to increase the retry counter. We could log it, but I find that it just adds noise, and we did log the first error.

---

Alright 👌🏻